### ISSUE TITLE
[REQUEST]: Extend Rust crates support to nightly and beta

### ISSUE THREAD

**Issue by ojeda** [compiler-explorer/compiler-explorer#3766]
### Is your feature request related to a problem? Please describe

The newly added Rust crates support do not seem to be available for nightly nor beta.

### Describe the solution you'd like

To have the same set of libraries available for nightly and beta.

### Describe alternatives you've considered

Not applicable

### Additional context

Not applicable

**Comment by partouf** [comment]
Right, sorry, but this was very much an intentional choice. We'll have to first have a feel in general how expensive the current situation is. Nightly building 100+ libraries might prove too costly.

**Comment by ojeda** [comment]
Yeah, I guessed it could be related to resource availability. It is still great to have it for the stable ones. Maybe later it could be introduced for a smaller set.

**Comment by canova** [comment]
Hi, I was wondering about crates support for nightly as well but I also understand the resource constraints.

> We'll have to first have a feel in general how expensive the current situation is. Nightly building 100+ libraries might prove too costly.

Do you know the general cost of this now? And do you think if it's manageable now?
I can't use libraries for my test case as some unstable features of rust requires nightly. Would love to have library support for nightly at least if not for beta.

Also just wanted to say that I love compiler explorer so thanks for this :)

**Comment by partouf** [comment]
It's beginning to be possible, but we need some extra things implemented, some of which we already have PR's but need testing and some still somewhere in the back of my mind but I will eventually remember, write down and fix.
So it will be a while, but probably still this year.
